### PR TITLE
Add Cairo update check workflow

### PR DESCRIPTION
**Pull request by mkaput**

### CODE REVIEW

File: .github/workflows/cairo-update-check.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: software-mansion/setup-scarb@v1
+        with:
+          scarb-version: nightly
```
Comment (piotmag769) [inline .github/workflows/cairo-update-check.yml:25]: Are we sure scarb will release nightly on daily basis? Why is Scarb even here?

File: .github/workflows/cairo-update-check.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      slack:
+        description: "Send Slack notification on failure"
+        type: boolean
+        default: true
+  schedule:
+    - cron: '0 0 * * *'
+
+concurrency:
+  group: ${{ github.workflow }}
+  cancel-in-progress: true
```
Comment (piotmag769) [inline .github/workflows/cairo-update-check.yml:15]: Why do we need it?

File: .github/workflows/cairo-update-check.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        run: |
+          CAIRO_REV=$(git ls-remote --refs "https://github.com/starkware-libs/cairo" main | awk '{print $1}')
+          echo "::notice::Checking Cairo commit: https://github.com/starkware-libs/cairo/commit/$CAIRO_REV"
+          cargo xtask set-cairo-version --rev "$CAIRO_REV"
+
+      - run: cargo test --profile=ci --no-fail-fast
+
+  notify_failed:
+    runs-on: ubuntu-latest
+    if: always() && !(inputs.slack) && needs.check.result == 'failure'
```
Comment (piotmag769) [inline .github/workflows/cairo-update-check.yml:40]: What does `always()` do? Why `!(inputs.slack)`?

File: .github/workflows/cairo-update-check.yml (hunk 2 above)
Comment (mkaput) [reply .github/workflows/cairo-update-check.yml:15]: This mostly came by copy-pasting, but in general for this workflow always the last run represents the most fresh state so we will never be interested in old runs, so why not?

File: .github/workflows/cairo-update-check.yml (hunk 3 above)
Comment (piotmag769) [reply .github/workflows/cairo-update-check.yml:40]: Yea but why the `!`?

File: .github/workflows/cairo-update-check.yml (hunk 1 above)
Comment (mkaput) [reply .github/workflows/cairo-update-check.yml:25]: discussed offline

File: .github/workflows/cairo-update-check.yml (hunk 3 above)
Comment (mkaput) [reply .github/workflows/cairo-update-check.yml:40]: 🤦🏻

**Review by piotmag769 — APPROVED** [review]

**Pull request merged**